R/tm_g_distribution.R

```diff
@@ -1200,39 +1223,58 @@ srv_distribution <- function(id,
         }
         qenv %>%
           # used to display table when running show-r-code code
-          teal.code::eval_code(quote(test_stats))
+          teal.code::eval_code(quote(test_table))
       }
     )
 
     # outputs ----
     ## building main qenv
-    output_q <- reactive({
-      tab <- input$tabs
-      req(tab) # tab is NULL upon app launch, hence will crash without this statement
-
-      qenv_final <- common_q()
+    output_common_q <- reactive({
       # wrapped in if since could lead into validate error - we do want to continue
-      test_r_qenv_out <- try(test_q(), silent = TRUE)
-      if (!inherits(test_r_qenv_out, c("try-error", "error"))) {
-        qenv_final <- c(qenv_final, test_q())
+      test_q_out <- try(test_q(), silent = TRUE)
+      if (!inherits(test_q_out, c("try-error", "error"))) {
+        c(common_q(), test_q_out)
+      } else {
+        common_q()
       }
+    })
+
+    output_dist_q <- reactive(c(output_common_q(), req(dist_q())))
+    output_qq_q <- reactive(c(output_common_q(), req(qq_q())))
+
+    decorated_output_dist_q <- srv_teal_transform_data(
+      "d_dist",
+      data = output_dist_q,
+      transformators = decorators
+    )
+
+    decorated_output_qq_q <- srv_teal_transform_data(
+      "d_qq",
+      data = output_qq_q,
+      transformators = decorators
+    )
 
-      qenv_final <- if (tab == "Histogram") {
-        req(dist_q())
-        c(qenv_final, dist_q())
+    decorated_output_q <- reactive({
+      tab <- req(input$tabs) # tab is NULL upon app launch, hence will crash without this statement
+      if (tab == "Histogram") {
+        decorated_output_dist_q()
       } else if (tab == "QQplot") {
-        req(qq_q())
-        c(qenv_final, qq_q())
+        decorated_output_qq_q()
       }
-      qenv_final
     })
 
-    dist_r <- reactive(dist_q()[["g"]])
+    dist_r <- reactive({
+      req(output_dist_q()) # Ensure original errors are displayed
+      decorated_output_dist_q()[["plot"]]
+    })
 
-    qq_r <- reactive(qq_q()[["g"]])
+    qq_r <- reactive({
+      req(output_qq_q()) # Ensure original errors are displayed
```

Comment:
*Topic to discuss*

Needed to prevent module's original errors to be lost during `teal_transform_module`

Alternative would be to add an option in `teal:::srv_teal_transform_data` that would pass errors during transform